src/arrayMethodJoin.js

```diff
@@ -4,9 +4,21 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i !== 0) {
+        str += separator;
+      };
+
+      if (this[i] !== undefined && this[i] !== null) {
+        str += this[i];
+      };
+    };
```

Comment:
```suggestion
    for (let i = 0; i < this.length; i++) {
      if (this[i] !== undefined && this[i] !== null) {
        str += this[i];
      };
      
      if (i < this.length -1) {
        str += separator;
      };
    };
```
In such case you will always add separator at the end of each iteration while it's not the last element